File: hyunjin/0 week/00.recursion/recursion.cpp
#include <bits/stdc++.h>
using namespace std;

int factorial(int n) {
	if (n == 1 || n == 0) return 1;
	return n * factorial(n - 1);
}

int fibonacci(int n) {
	cout << "fibonacci: " << n << '\n';
	if(n == 0 || n == 1) return n;
	return fibonacci(n - 1) + fibonacci(n - 2);
}

int n = 5;

int main() {
	cout << factorial(n) << " " << fibonacci(n) << '\n';

	return 0;
}

File: hyunjin/0 week/00.recursion/recursion.js
'use strict';

function factorial(n) {
  if (n === 0 || n === 1) return n;
  console.log('factorial', n);

  return n * factorial(n - 1);
}

function fibonacci(n) {
  if (n === 0 || n === 1) return n;
  console.log('fibonacci', n);

  return fibonacci(n - 1) + fibonacci(n - 2);
}

function main() {
  const num = 5;

  const facto = factorial(num);
  console.log('factorial result', facto);
  console.log('=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=');
  fibonacci(num);
}

main();